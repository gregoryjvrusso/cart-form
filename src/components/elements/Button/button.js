import React from "react";
import { string } from "prop-types";
import { Container } from "./button.styles";

const Button = ({ children }) => (
  <Container data-testid="button">{children}</Container>
);

Button.propTypes = {
  children: string.isRequired,
};

Button.displayName = "Button";

export default Button;
