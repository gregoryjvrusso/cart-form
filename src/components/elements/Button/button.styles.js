import styled from "styled-components";
import { theme } from "styled-tools";

export const Container = styled.button`
  background-color: ${theme(`colors.texts.red`)};
  border: none;
  border-radius: ${theme("spaces.2")}px;
  color: ${theme("colors.texts.white")};
  font-size: ${theme("fontSizes.3")}px;
  font-weight: ${theme("fontWeights.semibold")};
  letter-spacing: -0.01px;
  padding: ${theme("spaces.3")}px;
  text-align: center;
  width: 100%;
`;
