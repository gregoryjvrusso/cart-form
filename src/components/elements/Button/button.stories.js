import React from "react";
import { storiesOf } from "@storybook/react";
import { ThemeProvider } from "styled-components";
import theme from "../../../common/theme";

import Button from "../Button";

storiesOf("Elements/Button", module).add("default", () => (
  <ThemeProvider theme={theme}>
    <Button>Clique Aqui</Button>
  </ThemeProvider>
));
