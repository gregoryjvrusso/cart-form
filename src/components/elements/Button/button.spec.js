import React from "react";
import { render } from "@testing-library/react";
import Button from "../Button";

describe("Button component", () => {
  it("deve conseguir renderizar [snapshot]", async () => {
    const { container } = render(
      <Button>
        Clique aqui
      </Button>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
