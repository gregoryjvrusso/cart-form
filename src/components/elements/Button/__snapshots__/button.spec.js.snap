// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component deve conseguir renderizar [snapshot] 1`] = `
<button
  class="sc-AxjAm eEzdfU"
  data-testid="button"
>
  Clique aqui
</button>
`;
